File: gm-utils/src/main/java/src/commom/utils/object/PrimeiroNaoNulo.java
package src.commom.utils.object;

import gm.utils.lambda.F0;

public class PrimeiroNaoNulo {

	@SafeVarargs
	public static <T> T get(F0<T>... calls) {
		for (F0<T> ft : calls) {
			T o = ft.call();
			if (!Null.is(o)) {
				return o;
			}
		}
		return null;
	}

}

File: gm-utils/js/src/commom/utils/object/PrimeiroNaoNulo.js
import Null from './Null';

export default class PrimeiroNaoNulo {

	static get(... calls) {
		for (ft : calls) {
			let o = ft();
			if (!Null.is(o)) {
				return o;
			}
		}
		return null;
	}

}
